Firmware/tone04.c

```diff
@@ -41,20 +41,24 @@ void timer4_set_fan0(uint8_t duty)
 	if (duty == 0 || duty == 255)
 	{
 		// We use digital logic if the duty cycle is 0% or 100%
-		TCCR4A &= ~(1 << COM4C1);
-		OCR4C = 0;
-		WRITE(EXTRUDER_0_AUTO_FAN_PIN, duty);
+		ATOMIC_BLOCK(ATOMIC_RESTORESTATE)
+		{
+			TCCR4A &= ~(1 << COM4C1);
+			OCR4C = 0;
+			WRITE(EXTRUDER_0_AUTO_FAN_PIN, duty);
```

Comment:
the EXTRUDER_0_AUTO_FAN_PIN update can be moved out of the critical section

Reply:
Yes. I will move it now